sda/cmd/api/api_test.go

```diff
@@ -0,0 +1,509 @@
+		Repository: "rabbitmq",
+		Tag:        "3-management-alpine",
+	}, func(config *docker.HostConfig) {
+		// set AutoRemove to true so that stopped container goes away by itself
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+	})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	mqPort, _ = strconv.Atoi(rabbitmq.GetPort("5672/tcp"))
+	mqHostAndPort := rabbitmq.GetHostPort("15672/tcp")
+
+	client := http.Client{Timeout: 5 * time.Second}
+	req, err := http.NewRequest(http.MethodGet, "http://"+mqHostAndPort+"/api/users", http.NoBody)
+	if err != nil {
+		log.Fatal(err)
+	}
+	req.SetBasicAuth("guest", "guest")
+
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	if err := pool.Retry(func() error {
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		res.Body.Close()
+
+		return nil
+	}); err != nil {
+		if err := pool.Purge(postgres); err != nil {
+			log.Fatalf("Could not purge resource: %s", err)
+		}
+		if err := pool.Purge(rabbitmq); err != nil {
+			log.Fatalf("Could not purge resource: %s", err)
+		}
+		log.Fatalf("Could not connect to rabbitmq: %s", err)
+	}
+
+	RSAPath, _ := os.MkdirTemp("", "RSA")
+	if err := helper.CreateRSAkeys(RSAPath, RSAPath); err != nil {
+		log.Panic("Failed to create RSA keys")
+	}
+	ECPath, _ := os.MkdirTemp("", "EC")
+	if err := helper.CreateECkeys(ECPath, ECPath); err != nil {
+		log.Panic("Failed to create EC keys")
+	}
+
+	// OIDC container
+	oidc, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "python",
+		Tag:        "3.10-slim",
+		Cmd: []string{
+			"/bin/sh",
+			"-c",
+			"pip install --upgrade pip && pip install aiohttp Authlib joserfc requests && python -u /oidc.py",
+		},
+		ExposedPorts: []string{"8080"},
+		Mounts: []string{
+			fmt.Sprintf("%s/.github/integration/sda/oidc.py:/oidc.py", rootDir),
+		},
+	}, func(config *docker.HostConfig) {
+		// set AutoRemove to true so that stopped container goes away by itself
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+	})
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	OIDCport, _ = strconv.Atoi(oidc.GetPort("8080/tcp"))
+	OIDCHostAndPort := oidc.GetHostPort("8080/tcp")
+
+	client = http.Client{Timeout: 5 * time.Second}
+	req, err = http.NewRequest(http.MethodGet, "http://"+OIDCHostAndPort+"/jwk", http.NoBody)
+	if err != nil {
+		log.Panic(err)
+	}
+
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	if err := pool.Retry(func() error {
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		res.Body.Close()
+
+		return nil
+	}); err != nil {
+		if err := pool.Purge(oidc); err != nil {
+			log.Panicf("Could not purge oidc resource: %s", err)
+		}
+		log.Panicf("Could not connect to oidc: %s", err)
```

Comment:
```suggestion
		if err := pool.Purge(oidc); err != nil {
			log.Panicf("Could not purge oidc resource: %s", err)
		}
		if err := pool.Purge(postgres); err != nil {
			log.Fatalf("Could not purge resource: %s", err)
		}
		if err := pool.Purge(rabbitmq); err != nil {
			log.Fatalf("Could not purge resource: %s", err)
		}
		log.Panicf("Could not connect to oidc: %s", err)
```
The previously started containers needs to be removed as well.